File: spring project/animalshome/src/main/java/fmi/course/hcmi/animalshome/dto/PetAdWithUser.java
package fmi.course.hcmi.animalshome.dto;

import fmi.course.hcmi.animalshome.model.UserInfo;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
@Setter
public class PetAdWithUser {
    private PetAdDto petAdDto;
    private UserInfo owner;
}

File: angular project/Animals-Home/src/app/model/petAdWithUser.ts
import { PetAdDto } from './petAdDto';
import { UserInfo } from './userInfo';

export class PetAdWithUser {
    petAdDto: PetAdDto;
    owner: UserInfo;

    public setpetAdDto(petAdDto: PetAdDto) {
        this.petAdDto = petAdDto;
    }

    public setOwner(owner: UserInfo) {
        this.owner = owner;
    }
}